sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs

```diff
@@ -0,0 +1,179 @@
+
+using System.Diagnostics.CodeAnalysis;
+using System.ClientModel.Primitives;
+
+namespace System.ClientModel
+{
+    /// <summary>
+    /// Provides functionality to read and write <see cref="IPersistableModel{T}"/> and <see cref="IJsonModel{T}"/>.
+    /// </summary>
+    public static class ModelReaderWriter
+    {
+        /// <summary>
+        /// Converts the value of a model into a <see cref="BinaryData"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the value to write.</typeparam>
+        /// <param name="model">The model to convert.</param>
+        /// <param name="options">The <see cref="ModelReaderWriterOptions"/> to use.</param>
+        /// <returns>A <see cref="BinaryData"/> representation of the model in the <see cref="ModelReaderWriterOptions.Format"/> specified by the <paramref name="options"/>.</returns>
+        /// <exception cref="FormatException">If the model does not support the requested <see cref="ModelReaderWriterOptions.Format"/>.</exception>
+        /// <exception cref="ArgumentNullException">If <paramref name="model"/> is null.</exception>
+        public static BinaryData Write<T>(T model, ModelReaderWriterOptions? options = default) where T : IPersistableModel<T>
+        {
+            if (model is null)
+            {
+                throw new ArgumentNullException(nameof(model));
+            }
+
+            options ??= ModelReaderWriterOptions.Json;
+
+            if (options.Format == "J" || (options.Format == "W" && model.GetFormatFromOptions(options) == "J"))
+            {
+                if (model is not IJsonModel<T> jsonModel)
+                {
+                    throw new FormatException($"The model {model.GetType().Name} does not support '{options.Format}' format.");
+                }
+
+                using (ModelWriter<T> writer = new ModelWriter<T>(jsonModel, options))
+                {
+                    return writer.ToBinaryData();
+                }
+            }
+            else
+            {
+                return model.Write(options);
+            }
+        }
+
+        /// <summary>
+        /// Converts the value of a model into a <see cref="BinaryData"/>.
+        /// </summary>
+        /// <param name="model">The model to convert.</param>
+        /// <param name="options">The <see cref="ModelReaderWriterOptions"/> to use.</param>
+        /// <returns>A <see cref="BinaryData"/> representation of the model in the <see cref="ModelReaderWriterOptions.Format"/> specified by the <paramref name="options"/>.</returns>
+        /// <exception cref="InvalidOperationException">Throws if <paramref name="model"/> does not implement <see cref="IPersistableModel{T}"/>.</exception>
+        /// <exception cref="FormatException">If the model does not support the requested <see cref="ModelReaderWriterOptions.Format"/>.</exception>
+        /// <exception cref="ArgumentNullException">If <paramref name="model"/> is null.</exception>
```

Comment:
Should we address https://github.com/Azure/azure-sdk-for-net/issues/38244?

Are there other issues still open in https://github.com/Azure/azure-sdk-for-net/issues?q=is%3Aopen+is%3Aissue+label%3AAzure.Core that need to be addressed before releasing the types GA?

Reply:
I believe this is addressed.  Can you clarify what you think is missing?

Reply:
I do not see where we have listed `MissingMethodException` in the list of exceptions that can be thrown.  Is it no longer possible to throw this exception type, or am I missing where it is documented?